### PR TITLE
Add existing rules to validation report

### PR DESCRIPTION
**Pull request by zackt25**
As of today, Validation Rules (ie. Regex Validation rule / MinMax / Unique) are not reported into a "Data Compliance Report". The report is created only based on the audit methods.

The natural and typical user workflow could be:
- Create Validation Rules in a disabled state (do not enforce)
- Report for invalid objects, fix them
- Enable the rules once data discrepancies are understood and eventually fixed.

This PR is to add existing validation rules set by users and add it to the reporting document. 

I am adding a boolean var to the job file so that the user can select wether or not to include the existing validations that were created manually into the validation report.

### CODE REVIEW

**Comment by mzbroch** [comment]
@zackt25 

You need to remove `enabled=True` from `ValidationRuleManager` in `models.py`

**Comment by zackt25** [comment]
> @zackt25
> 
> I see no changes in models.py - You also need to remove `enabled=True` from `ValidationRuleManager` in `models.py`

It's been removed

File: nautobot_data_validation_engine/custom_validators.py [hunk 1]
```diff
@@ -146,6 +148,41 @@ def clean(self):
                     continue
                 compliance_class(self.context["object"]).clean()
 
+        if not exclude_disabled_rules:
+            if rule:
```
Comment (DistantVoyager) [inline nautobot_data_validation_engine/custom_validators.py:152]: When trying to run, I get the following error here:

image

```
ERROR/ForkPoolWorker-6] Task nautobot_data_validation_engine.jobs.RunRegisteredDataComplianceRules[e967e127-5b9e-40e2-9a91-ddbc40bcd96c] raised unexpected: UnboundLocalError("cannot access local variable 'rule' where it is not associated with a value")
nautobot-data-validation-engine-worker-1    | Traceback (most recent call last):
nautobot-data-validation-engine-worker-1    |   File "/usr/local/lib/python3.11/site-packages/celery/app/trace.py", line 477, in trace_task
nautobot-data-validation-engine-worker-1    |     R = retval = fun(*args, **kwargs)
nautobot-data-validation-engine-worker-1    |                  ^^^^^^^^^^^^^^^^^^^^
nautobot-data-validation-engine-worker-1    |   File "/usr/local/lib/python3.11/site-packages/nautobot/extras/jobs.py", line 148, in __call__
nautobot-data-validation-engine-worker-1    |     return self.run(*args, **deserialized_kwargs)
nautobot-data-validation-engine-worker-1    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
nautobot-data-validation-engine-worker-1    |   File "/source/nautobot_data_validation_engine/jobs.py", line 81, in run
nautobot-data-validation-engine-worker-1    |     self.report_for_validation_rules()
nautobot-data-validation-engine-worker-1    |   File "/source/nautobot_data_validation_engine/jobs.py", line 118, in report_for_validation_rules
nautobot-data-validation-engine-worker-1    |     validator(validated_object).clean(exclude_disabled_rules=False, logger=logger)
nautobot-data-validation-engine-worker-1    |   File "/source/nautobot_data_validation_engine/custom_validators.py", line 152, in clean
nautobot-data-validation-engine-worker-1    |     if rule:
nautobot-data-validation-engine-worker-1    |        ^^^^
nautobot-data-validation-engine-worker-1    | UnboundLocalError: cannot access local variable 'rule' where it is not associated with a value
```

Comment (zackt25) [reply nautobot_data_validation_engine/custom_validators.py]: @DistantVoyager It should be good now.

**Comment by mzbroch** [comment]
LGTM

File: nautobot_data_validation_engine/jobs.py [hunk 2]
```diff
@@ -47,6 +61,10 @@ class RunRegisteredDataComplianceRules(Job):
         description="Not selecting any rules will run all rules listed.",
     )
 
+    run_existing_rules_in_report = BooleanVar(
+        label="Run existing validation rules", description="Include manually created data validation rules in report?"
```
Comment (DistantVoyager) [inline nautobot_data_validation_engine/jobs.py:65]: Let's refer to the regex, min/max, unique, required rules as "built-in" rules

```suggestion
        label="Run built-in validation rules", description="Include created built-in data validation rules in report?"
```

Comment (zackt25) [reply nautobot_data_validation_engine/jobs.py]: fixed

**Comment by zackt25** [comment]
@DistantVoyager and @mzbroch 

> I'm still not able to get the expected outcome.
> 
> For example, I have a device named `bkk01-edge-01-badnameeee`, then create a regex rule conforming to an example like `aaa00-descriptor-01`. But after running the data compliance job with the existing rule button checked, it doesn't seem to flag that device as having a non-compliant device name.
> 
> image
> 
> But you can see here that the regex rule is catching it because it won't let me edit and save the device:
> 
> image

So this might be something to talk about. This code will only look at the rules that are disabled. 

Thinking was these steps:

1. SSoT brings in dirty data (rules are disabled)
2. Validation is done with disabled rules. 
3. Fixes are done on source SSoT side.
4. SSoT brings in good data. 
5. Nautobot is SOT now with the data and the rules can be enabled so any thing manually put in is checked.
or 
6. Keep the rules disabled and SSoT data keeps coming in and validated after without enabling rules.

**Comment by mzbroch** [comment]
> So this might be something to talk about. This code will only look at the rules that are disabled.

@DistantVoyager Sorry for the confusion - intention was as follow:
- live validation of objects is enforced only for "enabled" rules
- reporting is both for disabled and enabled rules (this is needed as in real life engineers will have data cleansing period before the rule is enforced and need to be reported)

File: nautobot_data_validation_engine/custom_validators.py [hunk 3]
```diff
@@ -146,6 +148,37 @@ def clean(self):
...
+
+        # Update `__all__` Data Compliance for given content type and object.
+        if not valid:
+            DataCompliance.objects.filter(
+                compliance_class_name=f"{instance._meta.model_name.capitalize()}DataComplianceRules",
+                content_type=ContentType.objects.get_for_model(instance),
+                object_id=instance.id,
+                validated_attribute="__all__",
+                valid=True,
+            ).update(valid=False)
```
Comment (DistantVoyager) [inline nautobot_data_validation_engine/custom_validators.py:164]: Need to update the message text as well:

image

**Review by DistantVoyager — APPROVED** [review]

**Pull request merged**